### ISSUE TITLE
Model.mode == tf.estimator.ModeKeys.TRAIN

### ISSUE THREAD

**Issue by lyzKF** [pytorch/fairseq#1776]
## Model.mode
```python
hparams.mode == tf.estimator.ModeKeys.TRAIN
```
**Is there a similar function in Fairseq ?**

**Comment by myleott** [comment]
Every PyTorch nn.Module has a `training` flag, so in a model you can usually access `self.training` to see if you're in training mode or not.

For example, look at how we use the functional form of dropout: https://github.com/pytorch/fairseq/blob/adb5b9c71f7ef4fe2f258e0da102d819ab9920ef/fairseq/models/lstm.py#L250

**Comment by lyzKF** [comment]
@myleott Thank you !

**Comment by lyzKF** [comment]
@myleott  Can i export the 'Transformer model' to be ONNX format in Fairseq?  if so, how should i do it?
